docs/DeveloperGuide.md

```diff
@@ -6,33 +6,230 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### The Overall UML Diagram:
```

Comment:
<img width="114" alt="image" src="https://github.com/nus-cs2113-AY2324S2/tp/assets/86831302/7f0bbbcf-66a9-496c-be6d-9bfb9bacbc6e">
Nice jobs on your uml diagram, but you can replace the red square and green circle with "+" and "-" sign.